### PR TITLE
feat:  Docker lint plugin

### LINKED ISSUE #5931: Feature: Add a plugin to support Dockerfile Linting in CI Pipeline
### 🔖 Feature description

Add Dockerfile Linting to CI Pipeline

### 🎤 Pitch / Usecases

I'd like you to please implement Dockerfile linting as part of the continuous integration (CI) pipeline to ensure that we are using best practices and coding standards. This feature will help improve the quality, security, and performance of Docker images used across the project by identifying common issues and recommending optimizations early in the development cycle.

### 🔄️ Alternative

NA

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/devtron-labs/devtron/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by kamal-devtron**
# Description

Fixes #5931 

## Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: scripts/sql/276_docker_lint.up.sql [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    arch=$(uname -m)
+    os=$(uname -s)
+    echo $arch
+    echo $os
+    command=$(wget https://github.com/hadolint/hadolint/releases/download/v2.12.0/hadolint-$os-$arch)
+    echo $command
+
+    cp hadolint-Linux-x86_64 hadolint
+    chmod +x hadolint
+    if [[ $FAIL_ON_ERROR == "true" ]]
```
Comment (tayalrishabh96) [inline scripts/sql/276_docker_lint.up.sql:33]: variable `FAIL_ON_ERROR` not exported

File: scripts/sql/276_docker_lint.up.sql [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+
+
+
+INSERT INTO "plugin_step" ("id", "plugin_id","name","description","index","step_type","script_id","deleted", "created_on", "created_by", "updated_on", "updated_by")
+VALUES (nextval('id_seq_plugin_step'),(SELECT id FROM plugin_metadata WHERE plugin_version='1.0.0' and name='DOCKER LINT' and deleted= false),'Step 1','Step 1 - Triggering DOCKER LINT','1','INLINE',(SELECT last_value FROM id_seq_plugin_pipeline_script),'f','now()', 1, 'now()', 1);
+
+
+INSERT INTO "plugin_step_variable" ("id", "plugin_step_id", "name", "format", "description", "is_exposed", "allow_empty_value", "variable_type", "value_type","default_value", "variable_step_index", "deleted", "created_on", "created_by", "updated_on", "updated_by")
+VALUES (nextval('id_seq_plugin_step_variable'), (SELECT ps.id FROM plugin_metadata p inner JOIN plugin_step ps on ps.plugin_id=p.id WHERE p.plugin_version='1.0.0' and p.name='DOCKER LINT' and p.deleted=false and ps."index"=1 and ps.deleted=false), 'DOCKER_FILE_PATH','STRING','Specify the file path to the Dockerfile for linting or processing ',false,true,'INPUT','NEW','true',1 ,'f','now()', 1, 'now()', 1);
```
Comment (tayalrishabh96) [inline scripts/sql/276_docker_lint.up.sql:58]: - `is_exposed` is set as false and should be set as true
- `default_value` is set as true ?
- mention in description what will be the default value

File: scripts/sql/276_docker_lint.up.sql [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+
+INSERT INTO "plugin_step_variable" ("id", "plugin_step_id", "name", "format", "description", "is_exposed", "allow_empty_value","variable_type", "value_type","default_value", "variable_step_index", "deleted", "created_on", "created_by", "updated_on", "updated_by")
+VALUES (nextval('id_seq_plugin_step_variable'), (SELECT ps.id FROM plugin_metadata p inner JOIN plugin_step ps on ps.plugin_id=p.id WHERE p.plugin_version='1.0.0' and p.name='DOCKER LINT' and p.deleted=false and ps."index"=1 and ps.deleted=false), 'FAIL_ON_ERROR','STRING','Pass true/false to fail the pipeline or not ',true,false,'INPUT','NEW','false',1 ,'f','now()', 1, 'now()', 1);
```
Comment (tayalrishabh96) [inline scripts/sql/276_docker_lint.up.sql:62]: - mention in description what will be the default value

File: scripts/sql/276_docer_lint.down.sql [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+DELETE FROM plugin_step WHERE plugin_id = (SELECT id FROM plugin_metadata WHERE plugin_version='1.0.0' and name='DOCKER LINT' and deleted=false);
+
+
+DELETE FROM plugin_stage_mapping WHERE plugin_id =(SELECT id FROM plugin_metadata WHERE plugin_version='1.0.0' and name='DOCKER LINT' and deleted=false);
+
+
+DELETE FROM plugin_tag_relation WHERE plugin_id =(SELECT id FROM plugin_metadata WHERE plugin_version='1.0.0' and name='DOCKER LINT' and deleted=false);
+
+
+DELETE FROM pipeline_stage_step where ref_plugin_id in (SELECT id from plugin_metadata WHERE plugin_version='1.0.0' and name ='DOCKER LINT' and deleted=false);
```
Comment (tayalrishabh96) [inline scripts/sql/276_docer_lint.down.sql:13]: Down fails with below error :  
`ERROR:  update or delete on table "pipeline_stage_step" violates foreign key constraint "pipeline_stage_step_variable_pipeline_stage_step_id_fkey" on table "pipeline_stage_step_variable"
DETAIL:  Key (id)=(1) is still referenced from table "pipeline_stage_step_variable".`

**Review by tayalrishabh96 — DISMISSED** [review]
Tested all the scenarios - looks good to merge!

**Review by pawan-mehta-dt — DISMISSED** [review]

**Review by prakarsh-dt — APPROVED** [review]

**Pull request merged**